### PR TITLE
`ProductPurchaseFeaturesList`: refactor tests to `@testing-library/react`

### PR DESCRIPTION
**Pull request by flootr**
#### Proposed Changes

* `ProductPurchaseFeaturesList`: refactor tests to `@testing-library/react`

#### Testing Instructions

* Verify unit tests still pass

Related to #63409

### CODE REVIEW

File: client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx
Comment (tyxla) [inline client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx]: Now that we're touching this line, we could omit the redundant `={ true }`

```suggestion
			<ProductPurchaseFeaturesList { ...props } isPlaceholder />
```

File: client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx
Comment (tyxla) [inline client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx:91]: Can we get a bit more specific than that? Same for the similar ones below.

**Review by tyxla — COMMENTED** [review]
This generally looks good, considering we translated existing tests. 

I just wonder if we can get a bit more specific with those `.not.toBeEmptyDOMElement()`, since anything we render will make the test pass.

File: client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx
Comment (flootr) [reply client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx:91]: I think the main ingredient of this test is to see whether those spies got called / not called. Those function return a bunch of different components to render which I'm not exactly sure would be the best way to test (e.g. go through one by one). Would a snapshot test secure us from unexpected changes there?

Comment (tyxla) [reply client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx:91]: I guess snapshot tests here would be on the opposite side of the spectrum: they would break even if we move features around, which doesn't seem to be desired. 

I think there's a bunch of buttons rendered by this component, each feature having at least one button - perhaps we can ensure that a button is rendered, and add a comment that we're not looking for a particular feature, but _any_ feature really? That would give _some_ specificity while maintaining the original idea.

Or, alternatively, if we only care about the spies and whether they have been called, we could remove this assertion altogether.

Comment (flootr) [reply client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx:91]: If we want to move features around without breaking tests I'm unsure about the reasoning of testing what features we render. The original tests made sure there is _something_ rendered (not specified what) which is still the case. I guess one could argue there isn't much value in this assertion alone but more in conjunction with making sure the correct methods are called for a plan.

I'd like to make another call for snapshot tests to ensure we don't remove anything unexpected from those render methods. If a feature gets removed intentionally and we'd consider this a "breaking change" (do we?) then updating snapshots tests seems feasible to me (and it's really just hitting _u_ while running tests) but if something gets removed unintentionally those tests got us covered.

By the way, I think those `getXFeatures` methods could be refactored to be components of their own which would make this one more lightweight.

Comment (flootr) [reply client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx:91]: Actually, I have to think about it more. Would

> Or, alternatively, if we only care about the spies and whether they have been called, we could remove this assertion altogether.

work for you in the meantime?

Comment (tyxla) [reply client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx:91]: I understand and mostly agree with your rationale about snapshot tests. Removing the assertion would also work, of course, although less useful. 

I'm leaning more towards snapshot tests with your rationale above, but I'll leave the final decision to you.

Comment (flootr) [reply client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx:91]: I've not come to a good conclusion on this and I'm not convinced snapshot tests will cover my original intention about testing those methods. It should be possible to adjust e.g. wording in one of those feature components without breaking snapshot tests in this one. We could mock out those feature components and test for their presence individually (e.g. `screen.getByTestId( 'feature-custom-css' )`) but I'm unsure about the value of the effort. I think testing what those render would lean more in the space of those components and leaving the space of this one.

For now, I'd like to opt for keeping those assertions as they are (make sure the correct methods are called and that they return something without making conclusions about what exactly is rendered) to unblock the RTL migration.

Comment (flootr) [reply client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx:91]: @tyxla let me know if ^ works for you.

Comment (tyxla) [reply client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx:91]: Let's go with that 👍 We can iterate as needed, let's not focus on making this perfect, but rather on our goal, which is, as you said, migrating to RTL.

**Review by tyxla — APPROVED** [review]
Let's 🚢 🚀 

We can always iterate and improve in subsequent PRs if we find value in that.

File: client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx
Comment (tyxla) [inline client/blocks/product-purchase-features-list/test/product-purchase-features-list.jsx]:
```suggestion
		const { container } = render( <ProductPurchaseFeaturesList { ...props } isPlaceholder /> );
```

**Pull request merged**
